dot/core/service_test.go

```diff
@@ -555,22 +560,49 @@ func Test_Service_maintainTransactionPool(t *testing.T) {
 			Propagate: true,
 		}
 
-		extrinsic := types.Extrinsic{21}
-		vt := transaction.NewValidTransaction(extrinsic, validity)
+		ext := types.Extrinsic{21}
+		externalExt := types.Extrinsic(bytes.Join([][]byte{
+			{byte(types.TxnExternal)},
+			ext,
+			testHeader.StateRoot.ToBytes(),
+		}, nil))
+		vt := transaction.NewValidTransaction(ext, validity)
 
 		ctrl := gomock.NewController(t)
 		runtimeMock := NewMockRuntimeInstance(ctrl)
-		runtimeMock.EXPECT().ValidateTransaction(types.Extrinsic{21}).Return(nil, errTestDummyError)
+		runtimeMock.EXPECT().ValidateTransaction(externalExt).Return(nil, errTestDummyError)
+		runtimeMock.EXPECT().Version().Return(runtime.Version{
+			SpecName:         []byte("polkadot"),
+			ImplName:         []byte("parity-polkadot"),
+			AuthoringVersion: authoringVersion,
+			SpecVersion:      specVersion,
+			ImplVersion:      implVersion,
+			APIItems: []runtime.APIItem{{
+				Name: [8]byte{0xd2, 0xbc, 0x98, 0x97, 0xee, 0xd0, 0x8f, 0x15},
+				Ver:  3,
+			}},
+			TransactionVersion: transactionVersion,
+			StateVersion:       stateVersion,
+		})
+		runtimeMock.EXPECT().SetContextStorage(&rtstorage.TrieState{})
+
 		mockTxnState := NewMockTransactionState(ctrl)
 		mockTxnState.EXPECT().RemoveExtrinsic(types.Extrinsic{21}).Times(2)
 		mockTxnState.EXPECT().PendingInPool().Return([]*transaction.ValidTransaction{vt})
 		mockBlockState := NewMockBlockState(ctrl)
-		mockBlockState.EXPECT().GetRuntime(nil).Return(runtimeMock, nil)
+		mockBlockState.EXPECT().GetRuntime(&common.Hash{}).Return(runtimeMock, nil)
+		mockBlockState.EXPECT().BestBlockHash().Return(common.Hash{})
+
+		mockStorageState := NewMockStorageState(ctrl)
+		mockStorageState.EXPECT().TrieState(&common.Hash{}).Return(&rtstorage.TrieState{}, nil)
+		mockStorageState.EXPECT().GetStateRootFromBlock(&common.Hash{}).Return(&common.Hash{}, nil)
 		service := &Service{
 			transactionState: mockTxnState,
 			blockState:       mockBlockState,
+			storageState:     mockStorageState,
 		}
-		service.maintainTransactionPool(&block)
+		err := service.maintainTransactionPool(&block, common.Hash{})
```

Comment:
nit you should pass in a non-zero value i.e. `common.Hash{1}` to make sure it gets passed around correctly (without reset/empty value passed in)